contract/src/main.rs

```diff
@@ -989,6 +1068,29 @@ pub extern "C" fn transfer() {
         Option::<Key>::None,
     );
 
+    let events_mode: u8 = get_stored_value_with_user_errors(
+        crate::constants::EVENTS_MODE,
+        NFTCoreError::MissingEventMode,
+        NFTCoreError::InvalidEventMode,
+    );
+
+    if events_mode != 0 {
+        events::record_event(match events_mode {
+            3 => Event::Cep47Dict(CEP47Event::Transfer {
+                sender: token_owner_key,
+                recipient: target_owner_key,
+                token_id: token_identifier.clone(),
```

Comment:
Hi, the only non fundamental comment I had, not sure if useful, because I faced the same issue with token_identifier, is that in I used a reference here instead of cloning for later use of token_identifier in the code.

https://github.com/casper-ecosystem/cep-78-enhanced-nft/pull/136/files#diff-e6b367605eb370c608fd74585147b016fc1e3b7fbd190772df677f70c77a3302R9

Reply:
As I read that code you do clone it, but do it in `record_event(event_enum: Event)` instead of one level above, am I reading it incorrectly?

Reply:
That is correct but I don't say that to avoid a clone, I say that to avoid forcing cloning into main contract, just like a courtesy for later usage like

`let token_number = utils::get_token_index(&token_identifier);`

basically I am avoiding to force cloning on interface usage and nest the cloning into the interface. Not sure if I am making sense, just a small comment.